compute_endpoint/globus_compute_endpoint/cli.py

```diff
@@ -675,6 +675,7 @@ def _do_start_endpoint(
         try:
             if config_str is not None:
                 ep_config = load_config_yaml(config_str)
+                ep_config.source_content = config_str
             else:
                 ep_config = get_config(ep_dir)
 
```

Comment:
Hmm; a mild abstraction leak that only one path needs to set the `.source_content` field.  Consider having `load_config_yaml()` set the `.source_content` field to be congruent with the behavior of `get_config()`, since they both return a `Config` object.

If an earlier function in the call stack (c.f., `get_config()`) decides to ~overwrite~ update that field, then "okay."

(Depending on how judicious you want to be, this might pave the way to make `.source_content` non-optional.)

Reply:
Yeah, that's how I originally wrote this change. I agree with your comments, so I'll set `config.source_content` in `load_config_yaml()` and `_load_config_py()`.